sqlserver/datadog_checks/sqlserver/queries.py

```diff
@@ -158,6 +95,160 @@
 }
 
 
+def get_query_ao_availability_groups(sqlserver_major_version):
+    """
+    Construct the sys.availability_groups QueryExecutor configuration based on the SQL Server major version
+
+    :params sqlserver_major_version: SQL Server major version (i.e. 2012, 2019, ...)
+    :return: a QueryExecutor query config object
+    """
+
+    # AG - sys.availability_groups
+    ag_column_definitions = {
+        'availability_group': {
+            'sql_column': 'AG.group_id AS availability_group',
+            'metric_definition': {'name': 'availability_group', 'type': 'tag'},
+        },
+        'availability_group_name': {
+            'sql_column': 'AG.name AS availability_group_name',
+            'metric_definition': {'name': 'availability_group_name', 'type': 'tag'},
+        },
+    }
+
+    # AR - sys.availability_replicas
+    ar_column_definitions = {
+        'replica_server_name': {
+            'sql_column': 'AR.replica_server_name',
+            'metric_definition': {'name': 'replica_server_name', 'type': 'tag'},
+        },
+        'failover_mode': {
+            'sql_column': 'LOWER(AR.failover_mode_desc) AS failover_mode_desc',
+            'metric_definition': {'name': 'failover_mode', 'type': 'tag'},
+        },
+        'availability_mode': {
+            'sql_column': 'LOWER(AR.availability_mode_desc) AS availability_mode_desc',
+            'metric_definition': {'name': 'availability_mode', 'type': 'tag'},
+        },
+    }
+
+    # ADC - sys.availability_databases_cluster
+    adc_column_definitions = {
+        'database_name': {
+            'sql_column': 'ADC.database_name',
+            'metric_definition': {'name': 'database_name', 'type': 'tag'},
+        },
+    }
+
+    # DRS - sys.dm_hadr_database_replica_states
+    drs_column_definitions = {
+        'replica_id': {
+            'sql_column': 'DRS.replica_id',
+            'metric_definition': {'name': 'replica_id', 'type': 'tag'},
+        },
+        'database_id': {
+            'sql_column': 'DRS.database_id',
+            'metric_definition': {'name': 'database_id', 'type': 'tag'},
+        },
+        'database_state': {
+            'sql_column': 'DRS.database_state',
+            'metric_definition': {'name': 'database_state', 'type': 'tag'},
+        },
+        'synchronization_state': {
+            'sql_column': 'LOWER(DRS.synchronization_state_desc) AS synchronization_state_desc',
+            'metric_definition': {'name': 'synchronization_state', 'type': 'tag'},
+        },
+        'ao_log_send_queue_size': {
+            'sql_column': '(DRS.log_send_queue_size * 1024) AS log_send_queue_size',
+            'metric_definition': {'name': 'ao.log_send_queue_size', 'type': 'gauge'},
+        },
+        'ao_log_send_rate': {
```

Comment:
Why not follow the same format as `get_query_file_stats`? It's less verbose and also consistent so anyone reading this will more easily understand what both functions are doing. 

```
'DRS.database_state': {'name': 'database_state', 'type': 'tag'},
'LOWER(DRS.synchronization_state_desc) AS synchronization_state_desc': {'name': 'synchronization_state', 'type': 'tag'},
'(DRS.log_send_queue_size * 1024) AS log_send_queue_size': {'name': 'ao.log_send_queue_size', 'type': 'gauge'},
```

Reply:
Whoops, you're totally right. Overcomplicated it, tyvm!